### DISCUSSION TITLE
Schema for Coverage Analysis Request

### DISCUSSION THREAD

**Discussion by dselva80** [eose-tools-org/eose-api#7] [General]
Should contain  

- list of satellites
- List of sample points/point grid
- Starting/ending times
- Type of analysis response requested

See https://github.com/code-lab-org/tatc/blob/main/app/tatc_app/coverage/schemas.py

**Comment by ptgrogan** [comment]
I added a prototype [CoverageRequest implementation](https://github.com/eose-tools-org/eose-api/blob/dev-tatc-analysis/src/eose/coverage.py) with an [example TAT-C implementation/demonstration](https://github.com/eose-tools-org/eose-api/blob/dev-tatc-analysis/src/tatc-coverage-example.ipynb).

I didn't add the inputs to select the type of request to be performed.

**Comment by vinay0000** [comment]
@ptgrogan could you please clarify where in the [example TAT-C implementation/demonstration](https://github.com/eose-tools-org/eose-api/blob/dev-tatc-analysis/src/tatc-coverage-example.ipynb) is the propagation being performed?

In the current architecture, it seems that the propagation details are hidden from the user, leading to the CoverageRequest using the default orbit propagator of TAT-C or OrbitPy, etc.

An alternative architecture would involve using the PropagationResponse as an input to the CoverageAnalysis Request. This approach would decouple the propagator selection from the coverage calculator selection. This could be especially useful for leveraging higher fidelity propagation results, such as those from GMAT, to perform coverage analysis with TAT-C or OrbitPy.

**Reply by ptgrogan** [reply]
The TAT-C implementation of coverage analysis is inside the `coverage_tatc` function. TAT-C does not use time-stepped propagation to perform coverage analysis. Rather, it uses a [find_events](https://rhodesmill.org/skyfield/api-satellites.html) method in Skyfield that looks for observation events (rise/culmination/set). It uses an optimization function to look for elevation angle maxima and then a root finding function to find the moments when a spacecraft passes above and below a minimum elevation angle condition around each maxima.

The TAT-C demo implementation for orbit propagation is [separate](https://github.com/eose-tools-org/eose-api/blob/dev-tatc-analysis/src/tatc-propagation-example.ipynb).

**Reply by ptgrogan** [reply]
I agree, though, that there is room to expand the CoverageAnalysisRequest to pass in additional parameters. TAT-C will not be able to process most other types of requests.
